tests/test_mail.py

```diff
@@ -0,0 +1,67 @@
+from aea.mail.base import Envelope, MailBox, InBox, OutBox
+
+from aea.protocols.base.message import Message
+from aea.protocols.oef.serialization import DEFAULT_OEF, OEFSerializer
+from aea.channels.oef.connection import OEFConnection
+from aea.channels.local.connection import LocalNode, OEFLocalConnection
+from aea.protocols.base.serialization import ProtobufSerializer, JSONSerializer
+from queue import Queue
+import pytest
+
+
+def test_envelope_initialisaztion() : 
+	msg = Message(content='hello')
+	message_bytes = ProtobufSerializer().encode(msg)
+	envelope = Envelope(to="Agent1", sender="Agent0", protocol_id="my_own_protocol", message=message_bytes)
+	
+	envelope.to = "Agent1"
```

Comment:
If you want to test the setters than maybe do:
assert envelope.to = "Agent1", "Cannot set to value on Envelope."

Similarly, for the others.